apollo-router/src/plugins/authorization/mod.rs

```diff
@@ -0,0 +1,272 @@
+//! Authorization plugin
+// With regards to ELv2 licensing, this entire file is license key functionality
+
+use std::ops::ControlFlow;
+
+use http::StatusCode;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use tower::BoxError;
+use tower::ServiceBuilder;
+use tower::ServiceExt;
+
+use crate::graphql;
+use crate::layers::ServiceBuilderExt;
+use crate::plugin::Plugin;
+use crate::plugin::PluginInit;
+use crate::plugins::authentication::APOLLO_AUTHENTICATION_JWT_CLAIMS;
+use crate::register_plugin;
+use crate::services::supergraph;
+
+/// Authorization plugin
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema)]
+struct Conf {
+    /// Reject unauthenticated requests
+    require_authentication: bool,
+}
+
+struct AuthorizationPlugin;
+
+#[async_trait::async_trait]
+impl Plugin for AuthorizationPlugin {
+    type Config = Conf;
+
+    async fn new(init: PluginInit<Self::Config>) -> Result<Self, BoxError> {
+        if !init.config.require_authentication {
+            return Err("require_authentication is disabled".into());
```

Comment:
I don't think we should throw an error in that case. If they want to disable it for a specific amount of time just to try something or if the configuration is codegen it might be annoying.

Reply:
I don't understand this logic. It seems that if the plugin is enabled then it will always error if `require_authentication` is not set. In which case why is it an option at all?

Reply:
because we might have other authorization systems in the future

Reply:
but you're right, it should not throw an error, it should just deactivate it

Reply:
fixed in [be443b0](https://github.com/apollographql/router/pull/3002/commits/be443b0a0de8bd76c72fbc04b3c0d27451d88e91)